1_slices_modules_and_objects/exercises/merge_sort/src/lib.rs

```diff
@@ -1,11 +1,31 @@
 /// Create a sorted Vec from a `given` slice
 pub fn merge_sort(given: &[i32]) -> Vec<i32> {
-    unimplemented!();
+    if given.len() <= 1 {
+        return Vec::from(given);
+    }
+
+    let splitted = given.split_at(given.len() / 2);
+    
+    merge(&merge_sort(splitted.0), &merge_sort(splitted.1))
```

Comment:
Good, clear and nice. If you want a bit more of a functional/rusty way, you could try matching on `given.len()` but I'm not sure the code would be clearer